janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geoshape.java

```diff
@@ -547,6 +548,16 @@ public Geoshape read(ScanBuffer buffer) {
             try {
                 return GeoshapeBinarySerializer.read(inputStream);
             } catch (IOException e) {
+                if (buffer instanceof ReadArrayBuffer && l == 1) {
+                    // retry using legacy point deserialization
+                    ((ReadArrayBuffer) buffer).movePositionTo(0);
+                    VariableLong.readPositive(buffer);
+                    try {
+                        return point(buffer.getFloat(), buffer.getFloat());
+                    } catch (Exception e2) {
+                        // throw original exception
+                    }
+                }
                 throw new RuntimeException("I/O exception reading geoshape");
```

Comment:
I took out this if and put all statements in the try-catch. I thought it makes sense to throw the original exception since in the case of an error I don't think it's possible to be sure whether or not the unserializable geoshape is a legacy point or not.